src/mono/dbi/cordb-chain.cpp

```diff
@@ -0,0 +1,154 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+// File: CORDB-CHAIN.CPP
+//
+
+#include <cordb-blocking-obj.h>
+#include <cordb-chain.h>
+#include <cordb-frame.h>
+#include <cordb-thread.h>
+#include <cordb.h>
+
+using namespace std;
+
+HRESULT CordbChainEnum::Next(ULONG celt, ICorDebugChain* chains[], ULONG* pceltFetched)
```

Comment:
It might be interesting to ask VS debugger if they still want to work with the chains API or they are ready to deprecate it in favor of solely using ICorDebugThread3::CreateStackwalk(). Stack chains was the original .NET implementation of debugger stackwalking circa ~2001 and then CreateStackWalk() was intended to be its better, simpler replacement circa 2008. VS adopted CreateStackWalk() for dump debugging fairly quickly but I don't recall if they ever moved the live debugging code path to use it.